pkg/asset/agent/oc.go

```diff
@@ -0,0 +1,48 @@
+package agent
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// ExecuteOC will execute an oc command.
+func ExecuteOC(pullSecret string, command []string) (string, error) {
+	// create registry config
+	ps, err := os.CreateTemp("", "registry-config")
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		ps.Close()
+		os.Remove(ps.Name())
+	}()
+	_, err = ps.Write([]byte(pullSecret))
+	if err != nil {
+		return "", err
+	}
+	// flush the buffer to ensure the file can be read
+	ps.Close()
+	registryConfig := "--registry-config=" + ps.Name()
+	command = append(command, registryConfig)
+	var stdoutBytes, stderrBytes bytes.Buffer
+	cmd := exec.Command(command[0], command[1:]...)
```

Comment:
gosec is still complaining, but now for spurious reasons.

Reply:
yeap.. added `// #nosec G204` should fix it 🤞